src/Accounts/Accounts/CommonModule/TelemetryProvider.cs

```diff
@@ -133,49 +137,50 @@ public virtual void Flush()
         /// <returns></returns>
         public virtual AzurePSQoSEvent CreateQosEvent(InvocationInfo invocationInfo, string parameterSetName, string correlationId, string processRecordId)
         {
-            var data = new AzurePSQoSEvent
+            var qosEvent = new AzurePSQoSEvent
             {
                 CommandName = invocationInfo?.MyCommand?.Name,
-                ModuleName = invocationInfo?.MyCommand?.ModuleName,
                 ModuleVersion = invocationInfo?.MyCommand?.Module?.Version?.ToString(),
                 SessionId = correlationId,
                 ParameterSetName = parameterSetName,
                 InvocationName = invocationInfo?.InvocationName,
                 InputFromPipeline = invocationInfo?.PipelineLength > 0
             };
 
-            data.CustomProperties.Add("PSPreviewVersion", "4.0.0");
-            data.CustomProperties.Add("UserAgent", "AzurePowershell/Az4.0.0-preview");
-            if(invocationInfo?.BoundParameters?.ContainsKey(SubscriptionIdString) == true)
+            // below is workaround that current invocationInfo only contains private module name. Trimming '.private' is a workaround for the time being.
+            const string privateModuleSuffix = ".private";
+            string moduleName = invocationInfo?.MyCommand?.ModuleName;
+            if (moduleName != null && moduleName.StartsWith("Az.") && moduleName.EndsWith(privateModuleSuffix))
             {
-                object rawValue = invocationInfo.BoundParameters[SubscriptionIdString];
-                string subscriptionId = null;
-                if(rawValue is string)
-                {
-                    subscriptionId = rawValue as string;
-                }
-                else if(rawValue is string[])
-                {
-                    string[] rawValueArray = rawValue as string[] ;
-                    if (rawValueArray.Length > 0)
-                    {
-                        subscriptionId = string.Join(";", rawValueArray);
-                    }
-                }
-                if(!string.IsNullOrEmpty(subscriptionId))
-                {
-                    data.CustomProperties.Add(SubscriptionIdString, subscriptionId);
-                }
+                moduleName = moduleName.Substring(0, moduleName.Length - privateModuleSuffix.Length);
             }
+            qosEvent.ModuleName = moduleName;
+
+
+            qosEvent.UserAgent = AzurePSCmdlet.UserAgent;
+            qosEvent.AzVersion = AzurePSCmdlet.AzVersion;
 
             if (invocationInfo != null)
             {
-                data.Parameters = string.Join(",", invocationInfo?.BoundParameters?.Keys?.ToArray());
+                qosEvent.Parameters = string.Join(" ",
```

Comment:
change the separator from `,` to ` ` on purpose?

Reply:
Yes, it makes generated cmdlet and handcrafted cmdlet have the same pattern
https://github.com/Azure/azure-powershell-common/blob/1a8cbb13e957e26789ec6b327b7b241bc0aaffbc/src/Common/AzurePSCmdlet.cs#L608